dev/src/transaction.ts

```diff
@@ -464,6 +463,25 @@ export class Transaction {
     );
     return Promise.reject(lastError);
   }
+
+  /**
+   * Delays further operations based on the provided error.
+   *
+   * @private
+   * @return A Promise that resolves after the delay expired.
+   */
+  private async maybeBackoff(error?: GoogleError) {
+    if (error) {
+      if (error.code === Status.RESOURCE_EXHAUSTED) {
+        this._backoff.resetToMax();
+      } else if (error.code === Status.ABORTED) {
+        // We don't backoff for ABORTED to avoid starvation
+        this._backoff.reset();
```

Comment:
This contradicts comments in go/transaction-retry-matrix. Do we have experimental data to support that starvation is a problem? Given multiple clients contending on a write, multiple clients failing to back off (with randomness) means they'll never make progress.

Reply:
I checked with the Firestore team and removed the special-casing on ABORTED, which simplifies the code a bit since we now always back off. This also allowed me to simplify the test suite.